crates/driver/src/boundary/score.rs

```diff
@@ -0,0 +1,59 @@
+use {
+    crate::{
+        domain::{
+            self,
+            competition::{self, score},
+            eth,
+        },
+        util::conv::u256::U256Ext,
+    },
+    solver::settlement_rater::{ScoreCalculator, ScoringError},
+};
+
+type SuccessProbability = f64;
+
+#[derive(Debug, Clone)]
+pub struct Score {
+    pub inner: ScoreCalculator,
+}
+
+impl Score {
+    pub fn new(score_cap: eth::U256, revert_protection: &domain::RevertProtection) -> Self {
+        Self {
+            inner: ScoreCalculator::new(
+                score_cap.to_big_rational(),
+                matches!(revert_protection, domain::RevertProtection::Disabled),
+            ),
+        }
+    }
+
+    pub fn score(
```

Comment:
Seems weird that you would have to call `Score::new()` and then `.score()` on something that is already called `Score`.
Wouldn't it make more sense to make the type a `ScoreCalculator(BoundaryScoreCalculator)`?

Reply:
Renamed to ScoreCalculator